### PR TITLE
feat(docs): clarifications, fixes #361

### LINKED ISSUE #361: Documentation overhaul (for clarity)
- People sometimes get confused by our `curl` commands, and there is also an inconsistency between placeholders in URL and non-URL (`:name` vs. `{token}`). Let's use braces style (consistent with OpenAPI)
- Under Windows, `curl` apparently works like this (cf. email to me on 2020-04-04):
  ```
    echo {"email": "youremailaddress@example.com", "password": "yourpassword"} | curl -X POST https://desec.io/api/v1/auth/login/ --header "Content-Type: application/json" --data @- 
  ```
  The user also sent a step-by-step instruction for Windows on 2020-04-07. Consider incorporating it somewhere.
- replace `password='[password]'` with `password='[TOKEN]'` to https://desec.readthedocs.io/en/latest/dyndns/configure.html#option-2-use-ddclient, and add "To get the Token: Login with your credentials defined in the registration process."

### PR DESCRIPTION
**Pull request by peterthomassen**

### CODE REVIEW

File: docs/auth/tokens.rst [hunk 1]
```diff
@@ -77,7 +77,7 @@ To delete an existing token by its ID via the token management endpoints, issue
 ``DELETE`` request on the token's endpoint, replacing ``:id`` with the
 token ``id`` value::
 
-    curl -X DELETE https://desec.io/api/v1/auth/tokens/:id/ \
+    curl -X DELETE https://desec.io/api/v1/auth/tokens/{id}/ \
```
Comment (nils-wisiol) [inline docs/auth/tokens.rst:80]: also in line 77

Comment (nils-wisiol) [reply docs/auth/tokens.rst:80]: maybe check kind of error for all the replacements

File: docs/dyndns/configure.rst [hunk 2]
```diff
@@ -32,11 +32,10 @@ following details in your router configuration:
...
 - Username (the full name of the domain you want to update, e.g. yourname.dedyn.io)
 - Hostname (same as your username)
-- Password (as provided when you registered your domain with us)
+- Token (long random string for authorization)
 
-**Advanced API users only:** The dynDNS password technically is an API token.
-If you also use our REST API, make sure to use a token for this purpose. Do not
-enter your account password when setting up your domain!
+**Advanced API users:** The dynDNS token technically is a regular API token.
+You can also use the token to make requests to our REST API.
```
Comment (nils-wisiol) [inline docs/dyndns/configure.rst:38]: add "Token scoping is on our roadmap."

File: docs/dyndns/lets-encrypt.rst [hunk 3]
```diff
@@ -31,8 +31,8 @@ steps.
    script, so that it can write the Let's Encrypt challenge to the DNS on your
    behalf. To do so, edit the ``.dedynauth`` file to look something like::
 
-    DEDYN_TOKEN=your token / dynDNS password
-    DEDYN_NAME=yourdomain.dedyn.io
+    DEDYN_TOKEN=[your token]  # remove brackets
+    DEDYN_NAME=[yourdomain].dedyn.io  # remove brackets
```
Comment (nils-wisiol) [inline docs/dyndns/lets-encrypt.rst:35]: maybe formatting can be used instead of "[]" to make this more clear?

File: docs/dyndns/update-api.rst [hunk 4]
```diff
@@ -20,23 +20,35 @@ connections.
 
 Authentication
 **************
-You can authenticate your client in several ways:
+You can authenticate your client in several ways. If authentication fails, the
+API will return a ``401 Unauthorized`` status code.
+
+Preferred method: HTTP Basic Authentication
+-------------------------------------------
+Encode your username and   token (provided during registration) in the
```
Comment (nils-wisiol) [inline docs/dyndns/update-api.rst:28]: spacing

File: docs/dyndns/update-api.rst [hunk 5]
```diff
@@ -20,23 +20,35 @@ connections.
 
 Authentication
 **************
-You can authenticate your client in several ways:
+You can authenticate your client in several ways. If authentication fails, the
+API will return a ``401 Unauthorized`` status code.
+
+Preferred method: HTTP Basic Authentication
```
Comment (nils-wisiol) [inline docs/dyndns/update-api.rst:26]: to be very obvious, maybe call this "HTTP Basic Token Authentication"

File: docs/dyndns/update-api.rst [hunk 6]
```diff
@@ -20,23 +20,35 @@ connections.
...
+---------------------------------
+Set the ``username`` and ``password`` query string parameters (``GET
+?username=...&password=...``).
+
+**Important:** We **strongly discourage** using this method as it comes with a
+subtle disadvantage: We log all HTTP request URLs for a few days to facilitate
+debugging. As a consequence, this method will cause your secret token to end
+up in our log files in clear text. The method is provided as an emergency
+solution where folks need to deal with old and/or crappy clients. If this is
+the case, we suggest looking for another client.
```
Comment (nils-wisiol) [inline docs/dyndns/update-api.rst:50]: Add: "This authentication method will be phased out" or "This authentication method will be phased out at the end of 2020"

File: docs/auth/tokens.rst (hunk 1 above)
Comment (peterthomassen) [reply docs/auth/tokens.rst:80]: yeah, I had been looking for occurrences of `/:` and miss all the other instances that were not preceded by a slash. Fixed them all now.

File: docs/dyndns/configure.rst (hunk 2 above)
Comment (peterthomassen) [reply docs/dyndns/configure.rst:38]: done

File: docs/dyndns/lets-encrypt.rst (hunk 3 above)
Comment (peterthomassen) [reply docs/dyndns/lets-encrypt.rst:35]: agreed not to (formatting inline in codeblocks is not straightforward)

File: docs/dyndns/update-api.rst (hunk 4 above)
Comment (peterthomassen) [reply docs/dyndns/update-api.rst:28]: done

File: docs/dyndns/update-api.rst (hunk 5 above)
Comment (peterthomassen) [reply docs/dyndns/update-api.rst:26]: done similarly

File: docs/dyndns/update-api.rst (hunk 6 above)
Comment (peterthomassen) [reply docs/dyndns/update-api.rst:50]: need more data before making this decision

**Pull request merged**